subsys/bluetooth/host/keys_br.c

```diff
@@ -52,31 +55,50 @@ struct bt_keys_link_key *bt_keys_get_link_key(const bt_addr_t *addr)
 	}
 
 	key = bt_keys_find_link_key(BT_ADDR_ANY);
-	if (key) {
-		bt_addr_copy(&key->addr, addr);
-		BT_DBG("created %p for %s", key, bt_addr_str(addr));
-
-		return key;
+	if (!key) {
+		if (IS_ENABLED(CONFIG_BT_KEYS_OVERWRITE_OLDEST)) {
```

Comment:
Why not use the aging counter scheme that the LE keys use? This seems like it would delete all keys from storage. Move it in the key_pool and not save it again.
Possibly don't support this Kconfig option just yet?